src/skadi/util.cc

```diff
@@ -71,7 +75,8 @@ weighted_grade(const std::vector<double>& heights,
   // Get the average weighted grade by homogenizing total weight. Return
   // max grades (up and down) and mean elevation.
   return std::make_tuple(total_grade * (1.0 / total_weight), max_up_grade, max_down_grade,
-                         (total_elev / n));
+                         (has_only_no_data ? static_cast<double>(baldr::kNoElevationData)
+                                           : total_elev / n));
```

Comment:
in this function we can check if the heights were actually all no_data. if so, we return our own no data value.

Reply:
i actually think we shouldnt change this, i think we should detect it on the other side of this call and convert it there. would you be cool with that? the reason being is that skadi is the interface to the raw data, baldr::kNoElevationData is a convention we use in our tile data. so the caller, somewhere in mjolnir should check if it got back a no data value and and convert it there imho

Reply:
Totally fine with me, it’s cleaner architecturally. Then I’ll return the raw no data value and let mjolnir detect and handle it